### ISSUE TITLE
Performance evaluation questions

### ISSUE THREAD

**Issue by wilsonwang371** [WasmEdge/WasmEdge#1693]
Hi guys,

I am running wasmedge together with wamr and wasmtime to compare the performance.

The bench mark that I used is https://github.com/attractivechaos/matmul

I compared AOT performance between them. What I got is a little bit strange. wasmtime and wasmedge performance is similar. But wamr has a better performance.

Can someone help me explain this?

```bash
$ time wasmedge ./mutmul.wasmedge.so -a 0 -n 3000
CPU time: 422.781
Central cell: 744.603

real        7m3.840s
user        7m2.583s
sys        0m0.544s



$ time wasmtime run ./mutmul.cwasm --allow-precompiled -- -a 0 -n 3000
CPU time: 0
Central cell: 744.603

real        7m0.237s
user        6m58.557s
sys        0m0.551s


$ time ~/wasm-micro-runtime/product-mini/platforms/linux/build/iwasm ./mutmul.wamr -n 3000 -a 0
CPU time: 286.265
Central cell: 744.603

real        4m47.114s
user        4m46.332s
sys        0m0.173s

```

**Comment by hydai** [comment]
Thanks for reporting this. @LFsWang Could you please help with this issue? Let's do some profiling on this benchmark to check what's going on here.

**Comment by LFsWang** [comment]
I try to reproduce on my PC. Is the argument same as yours?

## Compile 

```bash
emcc -O3 ./matmul/matmul.c -o mat.wasm -s TOTAL_MEMORY=671088640
```

## Run

### Wasmedge

```bash
wasmedgec mat.wasm mutmul.wasmedge.so
wasmedge ./mutmul.wasmedge.so -a 0 -n 3000
```

### Wasmtime

```bash
wasmtime compile mat.wasm
wasmtime run ./mat.cwasm --allow-precompiled -- -a 0 -n 3000
```

### Wasm-micro-runtime (Wamr)

```bash
$WSMRC -o mat.wamr mat.wasm
$IWSMR ./mat.wamr -a 0 -n 3000
```

## Result

Environment information

* CPU : AMD 5600X
* RAM : 32GB with 3200 MHz
* OS: Win 11 22H2 WSL 2 (Ubuntu 20.04.4 LTS)

||Wasmedge|Wasmtime|Wamr|
|-|-|-|-|
|1|22.59|27.247|22.725|
|2|22.6|26.389|22.602|
|3|22.641|26.928|22.462|
|4|22.57|26.542|22.462|
|5|22.674|27.385|22.694|
|6|22.529|26.866|22.241|
|7|22.646|27.217|22.362|
|8|22.537|27.935|22.402|
|9|22.579|27.739|22.523|
|10|22.606|27.539|22.514|
|Avg±sd (sec.)|22.60±0.05|27.18±0.5|22.5±0.15|


* note: Wasmedge raise the error below. But the result of the matrix is correct.

```
[error]     When executing function name: "_start"
```

**Comment by wilsonwang371** [comment]
Thanks! Let me take a look at this

**Comment by hydai** [comment]
@LFsWang Could you please provide the versions of runtimes you used for the above benchmarks?

**Comment by LFsWang** [comment]
* Wasmedge 0.10.1
* Wasmtime  0.39.1
* Wasm-micro-runtime (main 8/9)

**Comment by q82419** [comment]
> ```
> [error]     When executing function name: "_start"
> ```

Can be fixed by #1755

**Comment by dannypsnl** [comment]
Close for now, issue need a clear target, e.g. track long-term performance. That will be a CI progress to keep tracking.
